### PR TITLE
add feature "copy_aliases" for shrink action.

### PR DESCRIPTION
**Pull request by monkey3199**
This is the feature for Shrink action.
Aliases of the source index will be copied to the target index(shrunk index) before deleting the source index. So, you can search the shrink index with the same index alias. 

Copying is like below process.
1. Shrink target index
2. If you set 'copy_aliases' to true, then the aliases from the source index will be deleted and they will be added to the target index at the same time.
3. Delete the source index if you set 'delete_after' to true.

New option, 'copy_aliases' is added in option for shrink action. (default is false).
You can set the option in the action config file like action.yml.

This is the feature that I needed.
Thanks.

### CODE REVIEW

**Comment by karmi** [comment]
Hi @monkey3199, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/curator/pull/1060.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

File: examples/curator.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # not a Python "NoneType"
 client:
   hosts:
-    - 127.0.0.1
+    - 10.10.0.222
```
Comment (untergeek) [inline examples/curator.yml]: Why change this?  I don't understand.

**Comment by untergeek** [comment]
In addition to needing the CLA signed, tests (preferably integration tests) need to be added for this.  Is that something you're comfortable with?

File: examples/curator.yml (hunk 1 above)
Comment (monkey3199) [reply examples/curator.yml]: Sorry, That's only my configuration.

**Comment by monkey3199** [comment]
@karmi Maybe my email on the profile was hidden. I have set the email to public. Thanks.
@untergeek But, The tests for shrink is not in the source. Can I make the tests only for the feature "copy_aliases"?

**Comment by untergeek** [comment]
@monkey3199 OH SNAP! I can't believe I didn't do a `git add` on that file before I committed those changes and pushed.

Regardless, if you rebase on master now, [they're in there](https://github.com/elastic/curator/blob/master/test/integration/test_shrink.py).

File: examples/curator.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # not a Python "NoneType"
 client:
   hosts:
-    - 10.10.0.222
+  - 127.0.0.1
```
Comment (untergeek) [inline examples/curator.yml]: This still needs to be indented.

**Comment by untergeek** [comment]
@monkey3199 I can't merge this until you've signed the CLA...

**Comment by untergeek** [comment]
@monkey3199 I'm still waiting on that CLA getting signed.

**Comment by untergeek** [comment]
@monkey3199 I don't know what to do here.  I will have to close this PR if I do not get the CLA signed.  It's our official policy.

**Comment by monkey3199** [comment]
@untergeek I am so sorry. I'm going to get the CLA and let you know how it is going.

**Comment by monkey3199** [comment]
@untergeek I got the CLA again(email: monkey3199@gmail.com). And I will add the test soon.

**Comment by monkey3199** [comment]
I have just added the test code as well. I'm so sorry and apologize for being late. Please review this code.

File: test/integration/test_shrink.py [hunk 3]
```diff
@@ -179,4 +205,23 @@ def test_shrink_with_extras(self):
         self.assertTrue(settings[self.target]['settings']['index']['routing']['allocation'][allocation_type][key] == value)
         self.assertTrue(settings[self.idx]['settings']['index']['routing']['allocation']['require']['_name'] == '')
         self.assertEqual(settings[self.target]['settings']['index']['codec'], 'best_compression')
-        self.assertTrue(self.client.indices.exists_alias(index=self.target, name='my_alias'))
+        self.assertTrue(self.client.indices.exists_alias(index=self.target, name='my_alias')
```
Comment (untergeek) [inline test/integration/test_shrink.py]: You're missing the trailing `)` here, and that's why the test is failing.  It was there before, so you probably removed it by accident.

Comment (monkey3199) [reply test/integration/test_shrink.py]: Oh, sorry. I might delete when fixing some conflicts in merge. I will fix it.

**Comment by monkey3199** [comment]
I have added the line in the change log. Please confirm it.

**Comment by untergeek** [comment]
@monkey3199 Please make that last adjustment I requested in the review, and we're good to go.  I want to merge this in time to make it into 5.3

**Comment by monkey3199** [comment]
Thanks a lot!! Please let me know whatever I have to do.

**Comment by untergeek** [comment]
Check out my review.  Line 6 of `examples/curator.yml` should be indented the way it was.  That's all that remains.

**Comment by monkey3199** [comment]
@untergeek Oh! I have not been realizing about the review. I've just added the space words. Please check it again.

**Pull request merged**